src/matrix/storage/idb/stores/TimelineEventStore.ts

```diff
@@ -231,38 +261,38 @@ export class TimelineEventStore {
     }
 
     /** Inserts a new entry into the store. The combination of roomId and eventKey should not exist yet, or an error is thrown.
-     *  @param  {Entry} entry the entry to insert
-     *  @return {Promise<>} a promise resolving to undefined if the operation was successful, or a StorageError if not.
+     *  @param entry the entry to insert
+     *  @return a promise resolving to undefined if the operation was successful, or a StorageError if not.
      *  @throws {StorageError} ...
      */
-    insert(entry) {
+    insert(entry: StorageEntry): Promise<IDBValidKey> {
         entry.key = encodeKey(entry.roomId, entry.fragmentId, entry.eventIndex);
         entry.eventIdKey = encodeEventIdKey(entry.roomId, entry.event.event_id);
         // TODO: map error? or in idb/store?
-        this._timelineStore.add(entry);
+        return this._timelineStore.add(entry);
```

Comment:
So in addition to the earlier comment about not leaking `IDBValidKey` here, the lack of return is also sort off intentional. We don't have any store where keys are automatically created by indexeddb, so we never need the result of a `put` or `add`. Hence, we (I think) never await the promise, but rather make sure the transaction succeeds. The idea is that if there are a lot of `put`/`add`s, this creates a lot of `Promise` instances that are never awaited, so by not exposing it from the storage layer, I wanted to keep the option open of not even calling `reqAsPromise` on modification requests. I forgot the reason I haven't done so already.

Reply:
In short, perhaps don't add the returns for now and just make the return type `void`.

Reply:
I see now the jsdoc comment contradicts the above, but I just forgot to update it I assume.